BasicMusicPlayer/app/src/main/java/com/example/basicmusicplayer/MainActivity.kt

```diff
@@ -0,0 +1,92 @@
+package com.example.basicmusicplayer
+
+import android.media.AudioManager
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.webkit.WebResourceRequest
+import android.webkit.WebView
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import android.webkit.WebViewClient
+import android.media.MediaPlayer
+import android.media.AudioAttributes
+import android.provider.MediaStore.Audio
+import android.widget.ProgressBar
+import android.view.View
+import android.util.Log
+import android.widget.Button
+import android.widget.Toast
+import java.io.IOException
+
+//define main activity class and define properties
+class MainActivity : AppCompatActivity() {
+    private lateinit var loadingIndicator: ProgressBar;
+    private lateinit var btnPlayAudio : Button;
+    private lateinit var btnPauseAudio : Button;
+    private var mediaPlayer : MediaPlayer? = null;
```

Comment:
You should be able to make this property non-null if you initialize it at the property site. Something like this should work:
```
private val mediaPlayer = MediaPlayer().apply {
        setAudioAttributes(
            AudioAttributes.Builder()
                .setUsage(AudioAttributes.USAGE_MEDIA)
                .setContentType(AudioAttributes.CONTENT_TYPE_MUSIC)
                .build()
        )
        val wxycURL = "http://audio-mp3.ibiblio.org:8000/wxyc-alt.mp3"
        setDataSource(wxycURL)
    }


    init {
        mediaPlayer.setOnPreparedListener { mp ->
            // Called when the audio is prepared and ready to play
            mp.start()
            loadingIndicator.visibility = View.GONE // Hide the loading indicator
            Toast.makeText(this, "Audio started playing", Toast.LENGTH_LONG).show()
        }
    }
```

You could also initialize everything inside the `init` method, if you want to keep everything in a single block. You can't move the `setOnPreparedListener` to the property initialization block because it references `this`, which isn't fully initialized yet.

Reply:
Actually, you can move the `setOnPreparedListener` assignment into the property initializer if you change the context from `this` to `applicationContext` and get rid of `init` entirely.